### PR TITLE
Add support for google oauth2 provider.

### PR DESCRIPTION
**Pull request by haykbaluyan**
usage:

```
bin/trustyctl login --server https://localhost:7891 -r etc/dev/roots/trusty_dev_root_ca.pem --provider google
```

extra variables:

```
env | grep GOOGLE
TRUSTY_GOOGLE_CLIENT_ID=*****
TRUSTY_GOOGLE_CLIENT_SECRET=*****
```

```
....
test-runner_1   |               }
test-runner_1   |       ]
test-runner_1   | }
test-runner_1   | *** trusty intergation tests succeeded
trusty-ci_test-runner_1 exited with code 0
Aborting on container exit...
````

### CODE REVIEW

File: backend/service/auth/auth_service.go [hunk 1]
```diff
@@ -147,8 +179,8 @@ func (s *Service) AuthURLHandler() rest.Handle {
 			URL: conf.AuthCodeURL(base64.RawURLEncoding.EncodeToString(js), oauth2ResponseType, responseMode, nonce),
 		}
 
-		logger.Tracef("reqRedirectURL=%q, confRedirectURL=%q, deviceID=%s, url=%q",
-			redirectURL[0], conf.RedirectURL, deviceID[0], res.URL)
+		logger.Tracef("reqRedirectURL=%q, confRedirectURL=%q, deviceID=%s, oauth2Provider=%s, url=%q",
```
Comment (dissoupov) [inline backend/service/auth/auth_service.go:182]: just `provider`

File: backend/service/auth/auth_service.go [hunk 2]
```diff
@@ -344,13 +512,13 @@ func (s *Service) RefreshHandler() rest.Handle {
 // AuthDoneHandler handles v1.PathForAuthDone
 func (s *Service) AuthDoneHandler() rest.Handle {
 	return func(w http.ResponseWriter, r *http.Request, _ rest.Params) {
-		code, ok := r.URL.Query()["code"]
-		if !ok || len(code) != 1 || code[0] == "" {
-			marshal.WriteJSON(w, r, httperror.WithInvalidRequest("missing code parameter"))
+		token, ok := r.URL.Query()["token"]
```
Comment (dissoupov) [inline backend/service/auth/auth_service.go:515]: are you sure the Github is not broken with this change?

File: cli/auth/auth.go [hunk 3]
```diff
@@ -43,8 +44,12 @@ func Authenticate(c ctl.Control, p interface{}) error {
 		httpClient.WithTLS(tlscfg)
 	}
 
+	if flags.Oauth2Provider == nil || *flags.Oauth2Provider == "" {
+		return errors.New("please specify --oauth2-provider parameter")
```
Comment (dissoupov) [inline cli/auth/auth.go:48]: `--provider`

File: backend/service/auth/auth_service.go (hunk 2 above)
Comment (dissoupov) [reply backend/service/auth/auth_service.go:515]: i see in Google settings we use `response_type: code`
so it should be `code` and not `token` here

Comment (haykbaluyan) [reply backend/service/auth/auth_service.go:515]: nah,,, when AuthDoneHandler is called the Authorization Code flow is already passed and code is exchanged with token. Here we simply pass the token to the AuthDoneHandler. That is why you later call it TRUSTY_AUTH_TOKEN and not TRUSTY_AUTH_CODE.

**Review by dissoupov — APPROVED** [review]

**Review by dissoupov — APPROVED** [review]

**Pull request merged**